projects/ng-components-docs/src/app/examples/components/modals/modals.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, ViewChild } from "@angular/core";
+import { ModalSizeType, ModalPositionType } from "lib/src/modal/modal.config";
+import { ModalComponent as MC } from "lib/src/modal/modal.component";
+
+@Component({
+    selector: "app-modal",
+    templateUrl: "./modals.component.html",
+    styleUrls: ["./modals.component.scss"],
+})
+export class ModalComponent {
+    size: ModalSizeType;
+    center: boolean;
+    position: ModalPositionType;
+    fullscreen: boolean;
+    backdropDismiss: boolean = true;
+    @ViewChild(MC) modalChild: MC;
+
+    openModal(): void {
+        this.modalChild.open();
+    }
+
+    openSizeModal(size: ModalSizeType): void {
+        this.size = size;
+        this.openModal();
+    }
+
+    openWithoutBackdrop(): void {
+        this.backdropDismiss = false;
+        this.openModal();
+    }
+
+    openCenteredModal(): void {
+        this.center = true;
+        this.openModal();
+    }
+
+    openFullscreen(): void {
+        this.fullscreen = true;
+        this.openModal();
+    }
+
+    openAside(position: ModalPositionType) {
+        this.position = position;
+        this.openModal();
+    }
+
+    closeModal(): void {
+        this.modalChild.close();
+        setTimeout(() => {
+            this.resetModal();
+        }, 500);
```

Comment:
why the timer?

Reply:
the demo page will allow the user to trigger the same modal with different inputs, resetting the modal directly after trying to close it will affect the fade out animation, that is why i added the timer.
In normal scenarios the modal will trigger the same way instead of having it aside left then right then full screen on the same page and using the same modal

Reply:
this one is removed